packages/js/product-editor/src/blocks/section/edit.tsx

```diff
@@ -15,28 +17,91 @@ import {
  */
 import { sanitizeHTML } from '../../utils/sanitize-html';
 import { SectionBlockAttributes } from './types';
+import { hasAttributesUsedForVariations } from '../../utils';
 
 export function Edit( {
 	attributes,
-}: BlockEditProps< SectionBlockAttributes > ) {
+	context,
+}: BlockEditProps< SectionBlockAttributes > & {
+	context?: {
+		selectedTab?: string | null;
+	};
+} ) {
 	const { description, title, blockGap } = attributes;
+	const tab = context?.selectedTab || '';
 	const blockProps = useBlockProps();
+	const [ productAttributes ] = useEntityProp< Product[ 'attributes' ] >(
+		'postType',
+		'product',
+		'attributes'
+	);
+	const isDisabledSection =
+		hasAttributesUsedForVariations( productAttributes ) &&
+		[ 'inventory', 'pricing', 'shipping' ].includes( tab );
```

Comment:
I don't think that a generic section block should have specific knowledge of items outside of it, such as hardcoded tab (group) name/ids.

With the template API, fields that should be disabled could be in other locations of the template.

I think that a more robust solution ultimately would be to include the ability to disable fields conditionally on blocks as part of the conditional visibility work (see https://github.com/woocommerce/woocommerce/issues/39434).

I'll leave some more comments related to the current implementation in this PR, but ultimately I do not think that we should implement this directly in the generic section block.